Fix SLHC parameter for Run-2 emulation

dildick opened the pull request:
@davidlange6 @mark-grimes 
This was discussed around end October 2015 but was never fixed in the release

gregor-mittag committed:
Clean-up of PedeSteererWeakModeConstraints.

  Introduced some C++11 features to streamline the code.

gregor-mittag committed:
First version of multi-IOV weak-mode constraint.

 - Create constraint configurations for each IOV.
 - technically working: compiling and pede does not complain
   -> Need to verify that the result looks as expected
 - empty IOVs are problematic because the associated parameters are removed by
   pede's cut on the number of hits (since there are 0 hits)
   -> either ignore IOVs by providing a list of IOVs
      or use pede option 'skipemptycons' (preferred)

Dr15Jones committed:
Moved ProductHolderBase to its own header and source file

By removing the base class into its own header file we reduce
coupling to the different implementations of the inheriting classes.

Dr15Jones committed:
Consolidate code into PuttableProductHolder

ScheduledProductHolder and SourceProductHolder were identical except for their default status. Consolidated the two into PuttableProductHolder which is passed the default status at constructor time.

Dr15Jones committed:
Made methods of InputProductHolder private

There methods of InputProductHolder were only used by the class itself so they were made private.

Dr15Jones committed:
Removed unneeded methods of ProductHolders

Removed the explicit empty destructors for all ProductHolders since the default one is sufficient.
Removed the mergeProduct taking two arguments since it is never called.

gregor-mittag committed:
Apply constraint at the lowest defined level.

 - default (next level) behaviour can be changed now
 - added configuration flag to apply constraint at lowest level

Dr15Jones committed:
Made ProductData an internal detail of Principal

ProductData is no longer a class used outside of the internals of Principal.

Commit:
Fix SLHC parameter for Run-2 emulation

cmsbuild commented:
A new Pull Request was created by  @dildick (Sven Dildick) for CMSSW_8_0_X.

It involves the following packages:

L1Trigger/CSCTriggerPrimitives


@cmsbuild, @mulhearn, @davidlange6 can you please review it and eventually sign? Thanks.
@valuev, @Martin-Grunewald this is something you requested to watch as well.
@slava77, @degano, @smuzaffar you are the release manager for this.

cms-bot commands are list here cms-sw/cmssw#13028

slava77 commented on L1Trigger/CSCTriggerPrimitives/python/cscTriggerPrimitiveDigis_cfi.py:
```diff
@@ -11,7 +11,7 @@ def _modifyCscTriggerPrimitiveDigisForRun2( object ) :
     object.debugParameters = True
     object.checkBadChambers = False
-    object.commonParam.isSLHC = True
+    object.commonParam.isSLHC = False
```
make sure this doesn't propagate  to the upgrade scenarios.

dildick replied:
Good point. Since the "eras" are now available for upgrade scenarios, I can add appropriate  customization functions at the bottom of this file.

Commit:
Add era customizations

Commit:
Fix isSLHC

dildick commented:
@slava77 Are the customization functions in cscTriggerPrimitives_cfi sufficient?

slava77 commented:
@dildick it should be enough.

Dr15Jones committed:
Removed caching of product availability from InputProductHolder

Non of the other ProductHolders were caching their calculations of availability so this was removed to allow greater uniformity.

dildick commented:
Can you issue standard tests on this PR, please?

davidlange6 commented:
please test

> On Feb 16, 2016, at 9:47 PM, Sven Dildick <notifications@github.com> wrote:
> 
> Can you issue standard tests on this PR, please?
> 
> —
> Reply to this email directly or view it on GitHub.
>

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1 
Tested at: 82b0fdc344c61f9add60466001fd417f5cd70e03
 When I ran the RelVals I found an error in the following worklfows: 
 5.1 step1
<pre>runTheMatrix-results/5.1_TTbar+TTbarFS+HARVESTFS/step1_TTbar+TTbarFS+HARVESTFS.log

</pre>
4.22 step2
<pre>runTheMatrix-results/4.22_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC/step2_RunCosmics2011A+RunCosmics2011A+RECOCOSD+ALCACOSD+SKIMCOSD+HARVESTDC.log

</pre>
135.4 step1
<pre>runTheMatrix-results/135.4_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS/step1_ZEE_13+ZEEFS_13+HARVESTUP15FS+MINIAODMCUP15FS.log

</pre>
140.53 step2
<pre>runTheMatrix-results/140.53_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI/step2_RunHI2011+RunHI2011+RECOHID11+HARVESTDHI.log

</pre>
8.0 step2
<pre>runTheMatrix-results/8.0_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS/step2_BeamHalo+BeamHalo+DIGICOS+RECOCOS+ALCABH+HARVESTCOS.log

</pre>
1306.0 step2
<pre>runTheMatrix-results/1306.0_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15/step2_SingleMuPt1_UP15+SingleMuPt1_UP15+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
1000.0 step2
<pre>runTheMatrix-results/1000.0_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT/step2_RunMinBias2011A+RunMinBias2011A+TIER0+SKIMD+HARVESTDfst2+ALCASPLIT.log

</pre>
1001.0 step2
<pre>runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4.log

</pre>
1003.0 step2
<pre>runTheMatrix-results/1003.0_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM/step2_RunMinBias2012A+RunMinBias2012A+RECODDQM+HARVESTDDQM.log

</pre>
134.911 step3
<pre>runTheMatrix-results/134.911_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT/step3_RunSinglePh2015D+RunSinglePh2015D+HLTDR2_25ns+RECODR2_25nsreHLT+HARVESTDR2_25nsreHLT.log

</pre>
1330.0 step2
<pre>runTheMatrix-results/1330.0_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15/step2_ZMM_13+ZMM_13+DIGIUP15+RECOUP15+HARVESTUP15.log

</pre>
9.0 step2
<pre>runTheMatrix-results/9.0_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST/step2_Higgs200ChargedTaus+Higgs200ChargedTaus+DIGI+RECO+HARVEST.log

</pre>
25.0 step2
<pre>runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step2_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

</pre>
25202.0 step2
<pre>runTheMatrix-results/25202.0_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25/step2_TTbar_13+TTbar_13+DIGIUP15_PU25+RECOUP15_PU25+HARVESTUP15_PU25.log

</pre>
50202.0 step2
<pre>runTheMatrix-results/50202.0_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50/step2_TTbar_13+TTbar_13+DIGIUP15_PU50+RECOUP15_PU50+HARVESTUP15_PU50.log

</pre>

 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-13243/11257/summary.html

Dr15Jones committed:
Created new base class DataManagingProductHolder

DataManagingProductHolder is now the only class which directly owns the ProductData and keeps track of the status.

dildick commented:
All tests fail with the same reason: Run2_2017 cannot be used in an era customization function. Instead I should use a corresponding "2017" modifier specifically for the muon system. This does not seem to be present in 
https://github.com/cms-sw/cmssw/blob/CMSSW_8_0_X/Configuration/StandardSequences/python/Eras.py

Could you please advise how to proceed? Thanks.

Dr15Jones committed:
Remove used of ProductHolderBase::putProduct with provenance

All concrete implementations of ProductHolderBase::putProduct which took ProductProvenance as an argument always ignored the provenance. This is because provenance handling isn't done by the ProductHolders anymore. Therefore all calls to putProduct now just pass the Wrapper.

makortel committed:
Move PixelSeedMergerQuadruplets phase1TkCustoms customizations to the file

Phase1 pixel RECO configuration should be the default, so there is no
need to bring these changes in via customize/era.

makortel committed:
Migrate enabling of recopixelvertexing to trackingPhase1 era

makortel committed:
Migrate pixelTracks customization from phase1TkCustoms to trackingPhase1 era

makortel committed:
Migrate preDuplicateMergingDisplacedTracks customization from phase1TkCustoms to trackingPhase1 era

makortel committed:
Migrate ClusterShapeHitFilterESProducer and MeasurementTrackerEvent customizations from phase1TkCustoms to phase1Pixel era

makortel committed:
Migrate siPixelClusters customizations from phase1TkCustoms to phase1Pixel era

makortel committed:
PixelCPEGeneric is already the Run2 default

dildick commented:
Hi,
Could you please advise how to proceed? 
Thanks.

Dr15Jones committed:
Made DataManagingProductHolder::theStatus_ atomic

Dr15Jones committed:
Renamed ProductHolderBase::onDemand to onDemandWasNotRun

Renamed the function to attempt to better described what the function does.

mulhearn committed:
Change empty payloads from Error to Warning, and stop after 5

sarafiorendi committed:
move to stage2 L1

Sam-Harper committed:
EG L1S1 to L1S2 migration

sarafiorendi committed:
remove comments on old quality definitions

Dr15Jones committed:
Added method setFailedStatus

sarafiorendi committed:
migrate to stage2 L!

sarafiorendi committed:
add fill description

sarafiorendi committed:
change to be version 12 of L3MuonTrajectorySeed

Dr15Jones committed:
Use status to determine state of Products in ProductHolders

Use the status information to decide if the data product should be retrieved from storage or to run an unscheduled producer. All code that changes the Wrapper pointer now also has to update the status. The only exception is how ProductHolders work in SubProcesses (which will be fixed in a later commit)

Dr15Jones committed:
Use special ProductHolder for SubProcess

Instead of partially sharing internal ProductHolder data between the Principals of the parent process and the SubProcess, we now use a ParentProcessProductHolder in the SubProcess Principal which calls back to the parent process' ProductHolder.

Dr15Jones committed:
DataManagingProductHolder does not need connectTo

Only the ParentProcessProductHolder needs a working connectTo method.

avetisya committed:
Commenting out L1 stuff and fixing prescale index for 80X

Dr15Jones committed:
ProductHolder connectTo takes parent Principal

Changed connectTo in ProductHolderBase to get the Principal from the parent process. This is needed if an unscheduled execution from the parent is delayed until a call in a SubProcess.

ssekmen committed:
Move modules to L1T

ssekmen committed:
Move modules to L1T

ssekmen committed:
Move modules to L1T

ssekmen committed:
Move modules to L1T

ssekmen committed:
Move modules to L1T

Martin-Grunewald committed:
Add edm::Ptr support to BXVector

sarafiorendi committed:
fix include guard

ferencek committed:
protecting the default pat jet configuration from being modified inadvertently

ferencek committed:
removing redundant lines

ssekmen committed:
Adding L1T objects

ssekmen committed:
Adding L1T objects

ssekmen committed:
Adding L1T objects

blwiner committed:
fix max algo comparison

Martin-Grunewald committed:
L1TRawToDigi unpacker fixes

Martin-Grunewald committed:
Magic number 512

ssekmen committed:
deleting IsolatedL1TPixelTrackCandidate.h

ssekmen committed:
added HLTPixelIsolTrackL1TFilter as a module

ssekmen committed:
added IsolatedPixelTrackCandidateL1TProducer as a module

ssekmen committed:
recommitting fix

Martin-Grunewald committed:
Fix

ssekmen committed:
new version and checksum

ssekmen committed:
l1t jets to taus

ssekmen committed:
l1t jets to taus

ssekmen committed:
l1t jets to taus

ssekmen committed:
l1t jets to taus

ssekmen committed:
l1t jets to taus

ssekmen committed:
l1t jets to taus

ssekmen committed:
l1t jets to taus

ssekmen committed:
changed version to 14

Martin-Grunewald committed:
Remove excessive blank lines in error messages

Martin-Grunewald committed:
Remove excessive blank lines in error messages

hengne committed:
add missing 50ns lumisections in json file, and change json file name to avoid confusion

Martin-Grunewald committed:
Remove extra parameter

Martin-Grunewald committed:
Fix fillDescription defaults

Commit:
Do no call std::abs() on unsigned int

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Don't check if address is 0, check is collections contains item

Compilers can always assume that comparing reference to 0/NULL/nullptr
is always false. Reference is not a pointer. I believe, instead author
wanted to check if there are any items in the collection.

The following errors are reported by GCC (6.0.0, r233941):

    ElectroWeakAnalysis/ZMuMu/plugins/MuTriggerAnalyzer.cc:184:12: error: the compiler can assume that the address of 'toc' will never be NULL [-Werror=address]
    ElectroWeakAnalysis/ZMuMu/plugins/ZGoldenFilter.cc:116:15: error: the compiler can assume that the address of 'toc' will never be NULL [-Werror=address]

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Dr15Jones committed:
Removed use of product uniqueness checking

The check that a WrapperBase passed to the Principal is unique was not thread safe. In addition, the framework always creates new wrappers when a EDProducer puts a product. Therefore only a source could make this mistake.

davidlange6 committed:
Merge pull request #13617 from dmitrijus/nosr_80x

Fix a small bug in an online DQM input source

davidlange6 committed:
Merge pull request #13619 from dmitrijus/l1fix_80x

Fix a minor error in DQM/L1TMonitor BuildFile

davidlange6 committed:
Merge pull request #13632 from davidlt/gcc600-clean-analysis

Clean up ElectroWeakAnalysis (GCC 6)

Dr15Jones committed:
Moved code for mergeTheProduct to ProductHolders

In order to avoid having the base class ProductHolderBase have direct access to the product (which could lead to threading problems) the merging of data products from the same branch is now handled by the inheriting classes.

Dr15Jones committed:
Removed direct access to product from ProductHolderBase

For thread safety, the status of the product must be checked before accessing the product since the status operates as the synchronizer. Providing direct access to the product from the base class makes it too easy to miss the checks on the status.
Now only the internals of the inheriting classes have access to the product directly. All other access must go through resolveProduct.

dildick commented:
Hi,
Could you please advise how to proceed? 
Thanks.

cmsbuild committed:
Merge pull request #13627 from mbandrews/FixSRTaskBins

Fix binning in ZS filter SRTask plots

cmsbuild committed:
Merge pull request #13594 from VinInn/GSFfixes

Gsf cleanup and debug

Commit:
Remove l1amax_ and l1amin_ fields from EcalDumpRaw

These fields are not used within the class and are triggered compilation
errors (integer overflow).

The following errors are reported by GCC (6.0.0, r233941):

    EventFilter/EcalRawToDigi/plugins/EcalDumpRaw.cc:151:11: error: integer overflow in expression [-Werror=overflow]

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

cmsbuild committed:
Merge pull request #13551 from makortel/trackingValidation_v7

Update tracking MC validation scripts

cmsbuild committed:
Merge pull request #13544 from vkhristenko/CMSSW_8_1_X

HCAL DQM small fixed

cmsbuild committed:
Merge pull request #13502 from tanmaymudholkar/chisqr81X

New chi-squared plots

cmsbuild committed:
Merge pull request #13517 from makortel/fixTrackingOnlyDQM

Fix trackingOnly DQM

cmsbuild committed:
Merge pull request #13539 from jfernan2/DTDQMMonitorModule-fixes-for-PoA

Fixes to DT*Trigger classes to make them compatible with the DT PoA

Commit:
Do not use address as a pointer

Compiler can assume that address will never be 0/NULL/nullptr.

The following errors are reported by GCC (6.0.0, r233941):

    DataFormats/SiStripCommon/src/SiStripFecKey.cc:104:19: error: the compiler can assume that the address of 'fec_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:104:19: error: the compiler can assume that the address of 'fec_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:132:19: error: the compiler can assume that the address of 'fec_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:132:19: error: the compiler can assume that the address of 'fec_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:251:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:251:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:267:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFecKey.cc:267:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripHistoTitle.cc:33:20: error: the compiler can assume that the address of 'key_object' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripHistoTitle.cc:33:20: error: the compiler can assume that the address of 'key_object' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripKey.cc:57:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripKey.cc:57:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:106:19: error: the compiler can assume that the address of 'det_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:106:19: error: the compiler can assume that the address of 'det_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:126:19: error: the compiler can assume that the address of 'det_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:126:19: error: the compiler can assume that the address of 'det_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:154:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:154:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:166:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripDetKey.cc:166:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFedKey.cc:85:19: error: the compiler can assume that the address of 'fed_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFedKey.cc:85:19: error: the compiler can assume that the address of 'fed_key' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFedKey.cc:151:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFedKey.cc:151:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFedKey.cc:164:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]
    DataFormats/SiStripCommon/src/SiStripFedKey.cc:164:16: error: the compiler can assume that the address of 'input' will always evaluate to 'true' [-Werror=address]

These issues were also reported by Clang for a long time.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Add missing <random> header

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Add <cmath> and use std::fabs()

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

Commit:
Solve integer overlow

Adding something to std::numeric_limits<int>::max() will lead to integer
overflow and failed compilation unless the resulting type is implicitly or
explicitly converted to higher capacity type.

Tested with GCC 6.0.0 (r233941).

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

cmsbuild committed:
Merge pull request #13560 from rrabadan/StoN-mod-highest-val-81

StoN in module highest values

davidlange6 committed:
Merge pull request #13631 from davidlt/gcc600-l1-misc

Clean up for GCC 6 (<cmath> header, std::, __afs cache files)

davidlange6 committed:
Merge pull request #13611 from davidlt/gcc600-reco-std-accumulate

Add missing <numeric> include for std::accumulate

smuzaffar committed:
Revert "Fix a minor error in DQM/L1TMonitor BuildFile"

smuzaffar committed:
Merge pull request #13638 from cms-sw/revert-13619-l1fix_80x

Revert "Fix a minor error in DQM/L1TMonitor BuildFile"

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

ahinzmann committed:
add PU id wp to MiniAOD

Dr15Jones committed:
Moved getProductData method to be protected

The method is only needed by the class and classes which inherit from it.

gregor-mittag committed:
Declare 'seedOnlyFromAbove_' as 'int'.

This member variable of the AlignmentTrackSelector was wrongly declared as
boolean, causing the gcc 5.3.0 to complain about a comparison which can never be
true.

sarafiorendi committed:
remove check on BXCollection validity

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

davidlange6 committed:
Merge pull request #13579 from makortel/fixMTVforHI

Fix track validation for heavy ions

cmsbuild committed:
Merge pull request #13567 from lihux25/fix_topo_in_HcalTextCalibrations_8XX

Fix "withTopo" issue

cmsbuild committed:
Merge pull request #13475 from schneiml/booking-attribution

Improve print_trace in DQMStore

cmsbuild committed:
Merge pull request #13477 from makortel/trackingPhase1EraMigration

Migrate 2017 tracking to eras

cmsbuild committed:
Merge pull request #13500 from mbandrews/ChStat+Thresh81X

Add channel status + laser threshold

cmsbuild committed:
Merge pull request #13358 from mbandrews/removeTP

ECALbyLumi+Timing

davidlange6 commented:
@dildick - right, looks lost - lets work on the 81x PR - I'll restart the tests

dildick commented:
Shall I close this one?

davidlange6 commented:
no - I guess its a bug to fix so we want the change

Commit:
Regenerate DQMServices/Core/src/ROOTFilePB.proto

We are moving to Protocol Buffers 2.6.1 from 2.4.1. New version does not
have any issues compiling with GCC 6.0.0 (r233941). In addition to that
2.4.1 is a 5 year old release. Remember that Protocol Buffers provide
forward and backward compatibility for written data.

How files were regenerated?

    cd $CMSSW_BASE/src
    protoc --cpp_out=. DQMServices/Core/src/ROOTFilePB.proto

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

davidlt committed:
Merge pull request #13651 from davidlt/gcc600-protobuf-261-dqm-regen

Regenerate DQMServices/Core/src/ROOTFilePB.proto

Dr15Jones committed:
Consolidated ProductStatus enumerations

Several of the ProductStatus enumerations were designating nearly the same status. These were consolidated to share the same status.

Commit:
Making an application for TB simulation of HGCal

wmtan committed:
Fix setComplete() call

wmtan committed:
Add base class and fix fillBranches()

makortel committed:
Need to use only phase1Pixel era in customize

It is that one that dictates whether the already era-migrated
customizations need to be applied or not.

Commit:
Do not use constexpr with std::log in C++14 for ICC

C++14 standard prohibits library implementation to mark random C++
standard library functions as being `constexpr`. That's why this isn't
working with Clang and ICC in general.

It's a bit of gray area here. GCC most likely will lower  `std::log` to
`__builtin_log_*` intrinsic which are not covered by C++14 standard.
Thus it's `__builtin_log_*` which is `constexpr`, thus making it work
under C++14.

Signed-off-by: David Abdurachmanov <David.Abdurachmanov@cern.ch>

makortel committed:
Enable castor reco for 2017

cmsbuild committed:
Merge pull request #13643 from ghellwig/fix-gcc530-warning_seedOnlyFromAbove_81X

Declare 'seedOnlyFromAbove_' as 'int'.

davidlange6 committed:
Merge pull request #13634 from davidlt/gcc600-reco-misc-clean

Misc cleanup for GCC 6 (missing header, integer overflow, wrong address usage, ..)

davidlange6 committed:
Merge pull request #13630 from swang373/EMTF_DQM_dev_81X

EMTF DQM Module for CMSSW_8_1_X

mandrenguyen committed:
pre-declare cut values

boudoul committed:
config for 2023dev and 2023sim

boudoul committed:
2023sim and 2023dev in std sequence

Dr15Jones committed:
Moved common implementation to DataManagingProductHolder::resolveProductImpl

The majority of the code for resolveProduct_ was in common between the different inheriting classes. This was moved to a common function of the base class.

davidlange6 committed:
Merge pull request #13642 from ahinzmann/addPuIdWp81

Add PU id working points to MiniAOD 8_1_X

Dr15Jones committed:
Moved asserts in ProductHolder to more appropriate spots

davidlange6 committed:
Merge pull request #13662 from deguio/fixT0Query

add protection for T0 configuration query

boudoul committed:
XML for 2023dev and sim

boudoul committed:
building scenario

boudoul committed:
last version of the tilted xmls

boudoul committed:
last version of the flat tracker xmls

boudoul committed:
write detid for flat and tilted

boudoul committed:
fix typo

wmtan committed:
New Base Class

Dr15Jones committed:
Unified productUnavailable_ and renamed onDemandWasNotRun

The productUnavailable_ method was moved to the DataManagingProductHolder base class since the function could be made the same for all inheriting classes.
onDemandWasNotRun was renamed to unscheduledWasNotRun to better match the naming convention of the framework.

hugedata committed:
Changed the struct Count to avoid thread safety warnings

Changed struct Count in SiTrivialInduceChargeOnStrips
so that it is const during normal compilation.

Avoids thread safety warnings in the Static analyzers

Dr15Jones committed:
mergeProduct_ moved to ProductHolder base class

Since all the versions of mergeProduct_ from classes that inherit from DataManagingProductHolder could be made identical, the implementation was moved to the base class.
This was also the last place outside of DataManagingProductHolder which called getProductData so that function could now be made private.

wmtan committed:
Don't lose provenance

wmtan committed:
Avoid confusion

cmsbuild committed:
Merge pull request #13667 from mdhildreth/PreMix_Validation_DigiSimLink_switch_80X

Add automatic turn-off of tracker digiSimLink production in standard production mode

cmsbuild committed:
Merge pull request #13658 from davidlt/fix-icc-GsfBetheHeitlerUpdator

Do not use constexpr with std::log in C++14 for ICC

davidlange6 committed:
Merge pull request #13645 from cms-tsg-storm/Cummulative81X

Cummulative 80X PRs for 81X: (#13558 #13572 #13588 #13596 #13602 #13607 #13621)

davidlange6 committed:
Merge pull request #13372 from cms-met/METCorDev_Jan16_METSig

MET PAT tool update + update of MET significance

davidlange6 committed:
Merge pull request #13530 from bendavid/pythia8cfgrandom_80xV4

Changes needed for SUSY parameter scans

cmsbuild committed:
Merge pull request #13571 from degano/ecalsim_warn_81X

[SimCalorimetry/EcalElectronicsEmulation] Solve warning found by gcc 5.3.0.

davidlange6 committed:
Merge pull request #13370 from diguida/conddbV2-remove-V1-deps-6_rebased

Further cleanup of Condition access in AlCaDB, online DQM, Geometry, HLT, Simulation, and standard configuration

davidlange6 committed:
Merge pull request #13600 from wmtan/ConsumesInterfaceForOutputModules

Enforce consumes interface in output modules

alexeybaskakov committed:
only Filter, no analyzer

lgray committed:
Add accessor to get the longitudinal thickness of the silicon wafers to HGCalDDDConstants

ianna committed:
* Migrate to new ConDB configuration
* Skip HGCal test for run1&2

Dr15Jones committed:
Can not add products to forwarding ProductHolders

The ProductHolder classes which forward requests to other ProductHolders should never have their methods which put or merge data called.

hugedata committed:
Added Count constructor

Dr15Jones committed:
Move logic for put or merge into ProductHolder

Instead of having the Principal calling several ProductHolder functions in order to decide if a product should be put or merged, the logic was moved to be internal to the ProductHolder.

Commit:
Add scenario to help testing 2016 HCAL setup

Dr15Jones committed:
Type checking done internally by ProductHolder

Previously, just before each call to ProductHolderBase::put or putAndMerge there was a call to ProductHolderBase::checkType. The call was moved to be inside the functions instead.

davidlange6 committed:
Merge pull request #13671 from boudoul/workflowsp2_81X_v2

Workflows for 2023 including Phase2 Tracker (gensim step)

cmsbuild committed:
Merge pull request #13654 from bsunanda/Phase2-hgx42

bsunanda:Phase2-hgx42 Making an application for TB simulation of HGCal

Dr15Jones committed:
checkType only accessible by DataManagingProductHolder

Only the inheriting class needs to be able to call checkType.

Dr15Jones committed:
putOrMergeProduct logic now moved to inheriting class

By moving the putOrMergeProduct logic into DataManagingProductHolder it is possible to reduce the number of virtual functions from the base class which are not callable from the ProductHolders which forward calls to other ProductHolders.

Dr15Jones committed:
Moved throwProductDeletedException to inheriting class

Only the class DataManagingProductHolder now called throwProductDeletedException so the function was moved to that class.

ferencek committed:
Update RelVal input to CMSSW_8_0_0

ferencek committed:
synchronizing default taggers with the recent RECO sequence cleanup

ferencek committed:
added missing cfi files

ferencek committed:
added missing taggers

ferencek committed:
fixed typos

ferencek committed:
making legacy sequence more in synch with the new one

ferencek committed:
updated list of taggers available in PAT

Commit:
Takes care of half cell positions

slava77devel committed:
use labeled objects directly; do not wrap them in unnamed sequences

ianna committed:
* Remove obsolete parameter

davidlange6 committed:
Merge pull request #13677 from lgray/hgc_longitudinal_info

Add accessor to HGCalDDDConstants

cmsbuild committed:
Merge pull request #13520 from ianna/extended2016-scenario-scripts

2016 Geometry Scenario Scripts

mandrenguyen committed:
Added protection against non-existant modules

cmsdqm committed:
Fix DQM/L1TMonitor BuildFile

So that it generates proper plugins.
I hope it won't crash the IB this time.

Dr15Jones committed:
Fixed incorrect merge conflict resolution

After rebasing, the changes to resolve a merge conflict were not correct and would not compile. This fixes the problem.

Commit:
Move the cff files to Configuration/Geometry and name them as GeometryExtended2016dev...

Dr15Jones committed:
Renamed class ProductHolder to ProductResolver

After refactoring the ProductHolder class it was clear that the class was not holding a product but instead is used to resolve access to one.

Dr15Jones committed:
Removed unnecessary friends from ProductResolverBase

The friendship relationship was no longer necessary.

Dr15Jones committed:
Only InputProductResolver can call Principal::readFromSource

Restricting readFromSource to only come from InputProductResolver allows some of the checks in the call to be removed since they are already guaranteed.

Dr15Jones committed:
Only UnscheduledProductResolver can call Principal::unscheduledFill

Dr15Jones committed:
Removed unused function ProductData::connectTo

Dr15Jones committed:
Removed unused swap method from ProductResolvers

ferencek committed:
TagInfo collection names need to end with 'TagInfos' to avoid possible ambiguities in PAT

davidlange6 committed:
Merge pull request #13679 from vanbesien/Fixing_reference_plot_linking

Fixed the missing linking between plot and refplot (8_1_X)

davidlange6 committed:
Merge pull request #13681 from slava77/CMSSW_8_0_0_patch1/hackaroundPATCutParser-temp-cleanPatJets

delete loaded cleanPatJets after the temporary load if it was not around already

davidlange6 committed:
Merge pull request #13603 from mandrenguyen/customizePPrecoForHI81X

Added customizations to run pp reco on peripheral HI events

davidlange6 committed:
Merge pull request #13698 from Dr15Jones/refactorProductHolders_8_1_X

Refactored ProductHolders

davidlange6 committed:
Merge pull request #13606 from ferencek/ProtectPATJetDefaults_from-CMSSW_8_1_X_2016-03-04-2300

Protect default PAT jet configuration (81X)

davidlange6 committed:
Merge pull request #13672 from bouril/makeCountThreadsafe

Changed the struct Count to avoid thread safety warnings

davidlange6 committed:
Merge pull request #13703 from cms-l1t-offline/rekovic-hlt-802-seed-unpackedGT

Rekovic hlt 802 seed unpacked gt

Commit:
updating  TrackerGeometry in Geometry/TrackerGeometryBuilder to accommodate the final naming scheme for Phase2 Pixel detectors and corresponding update in Phase2Digitizer

ianna committed:
* No need to checkout packages locally
* Add tracker topology

ianna committed:
Merged fix-validation-script from repository ianna

Conflicts:
	Validation/Geometry/test/dddvsdb/runDDDvsDBGeometryValidation.sh

ianna committed:
Allow multiple scenario test

Commit:
TrackerTopology is accessed from TrackerTopologyRcd

ianna committed:
Sync materials in dev scenario with what is now in 2015 prod scenario.

hengne committed:
Merge remote-tracking branch 'official-cmssw/CMSSW_8_1_X' into from-CMSSW_8_1_0_pre1_relval_fixJson_rebase

Dr15Jones committed:
Pass non nullptr to std::unique_ptr guard

When using a std::unique_ptr as a guard variable one must use a non-nullptr value else the customized destructor will not be run.

Commit:
Complete the example of TB for HGCal

Commit:
Fix bug to avoid run-time error for Pythia8

wmtan committed:
Avoid unneeded retrievel of per event provenance

cmsbuild committed:
Merge pull request #13718 from Dr15Jones/fixGuardXrdRequestManager

Pass non nullptr to std::unique_ptr guard

cmsbuild committed:
Merge pull request #13707 from cms-l1t-offline/rekovic-hlt-802-seed-unpackedGT

Rekovic hlt 802 seed unpacked gt (pruning on top of PR 13703)

cmsbuild committed:
Merge pull request #13691 from dmitrijus/l1fix_try2_81x

Fix DQM/L1TMonitor BuildFile

cmsbuild committed:
Merge pull request #13688 from slava77/CMSSW_8_1_0_pre1/sign677/repackable

cleanup unnamed sequences in TrackingDQMSourceTier0*

cmsbuild committed:
Merge pull request #13687 from bsunanda/Phase2-hgx43

bsunanda:Phase2-hgx43 Takes care of half cell positions

cmsbuild committed:
Merge pull request #13393 from schneiml/dqm-tracker-residuals-improvements

DQM TrackerMonitorTrack improvements

cmsbuild committed:
Merge pull request #13716 from hengne/from-CMSSW_8_1_0_pre1_relval_fixJson_rebase

Fix re-reco golden json, adding 50ns lumi blocks coverage

davidlange6 committed:
Merge pull request #13666 from makortel/phase1RecoEraMigration

Migrate remaining 2017 reco customizations to era

cmsbuild committed:
Merge pull request #13680 from ianna/py-config-port

Migrate to CondDB Configuration

cmsbuild committed:
Merge pull request #13706 from suchandradutta/81X_Phase2TrackerDigitizer

updating  TrackerGeometry in Geometry/TrackerGeometryBuilder to accom…

cmsbuild committed:
Merge pull request #13714 from ianna/materials-in-dev-scenario-sync

Sync Materials in 2015 Dev Scenario

Dr15Jones committed:
Made EarlyDeleteHelper thread-safe

Switched to using atomic counters in order to avoid race conditions once we start using more than one thread per event.

Dr15Jones committed:
Using meaningful names in BranchToCount

Changed names from first,second to branch,count to allow better self documentation.

cmsbuild committed:
Merge pull request #13720 from bsunanda/Run2-alca37

bsunanda:Run2-alca37 Fix bug to avoid run-time error for Pythia8

cmsbuild committed:
Merge pull request #13566 from jasperlauwers/8_1_X_correct_QuadPFJet

Correct QuadPFJet pathnames

Martin-Grunewald committed:
Need unqiue cfi file names to avoid overwriting

wmtan committed:
Add comment

cmsbuild committed:
Merge pull request #13727 from Dr15Jones/fixGuardXrdRequestManager

Made EarlyDeleteHelper thread-safe

cmsbuild committed:
Merge pull request #13705 from ianna/fix-validation-script

Fix Validation Script

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

Martin-Grunewald committed:
Fix Hcal migration for ConfDB

Martin-Grunewald committed:
Iso stuff

Commit:
Remove unneeded selector

ianna committed:
* Add 2023dev and 2023sim scenarios dump

makortel committed:
Fix typo

alexeybaskakov committed:
cout removed

davidlange6 committed:
Merge pull request #13732 from Martin-Grunewald/FixMuonMigration81X

Need unqiue cfi file names in fillDescription to avoid overwriting in HLT (81X)

davidlange6 committed:
Merge pull request #13722 from wmtan/DoNotReadPerEventProvenanceIfNotNeeded

Avoid unneeded retrievel of per event provenance

davidlange6 committed:
Merge pull request #13695 from jsalfeld/80X_Add_2016_SpringMC_PileupScenario_v1

80 x add 2016 spring mc pileup scenario v1

ianna committed:
* Cleanup obsolete scenarios

Commit:
Change name TestBeam to HGCalTestBeam

davidlange6 committed:
Merge pull request #13735 from wmtan/RemoveUnneededSelectors

Remove unneeded selector

makortel committed:
Add unit test catching the typo

davidlange6 committed:
Merge pull request #13685 from cms-btv-pog/PATbtagUpdate_from-CMSSW_8_1_X_2016-03-10-1100

PAT btag update (81X)

cmsbuild committed:
Merge pull request #13710 from ghellwig/multi-iov_weak-mode-constraints_81X

Multi-IOV weak mode constraints

cmsbuild committed:
Merge pull request #13719 from bsunanda/Phase2-hgx44

bsunanda"Phase2-hgx44 Complete the example of TB for HGCal

cmsbuild committed:
Merge pull request #13729 from makortel/fixRectangularTrackingRegions

Fix RectangularEtaPhiTrackingRegion

cmsbuild committed:
Merge pull request #13737 from ianna/add-2023-scenarios

Add 2023 Scenarios

cmsbuild committed:
Merge pull request #13738 from makortel/fixTypoTypes

Fix typo in Types.py

davidlange6 committed:
Merge pull request #13746 from slava77/CMSSW_8_0_2/fevtLoadClean

locally named load of FEVT in hotlineSkims

cmsbuild committed:
Merge CMSSW_8_0_X into CMSSW_8_1_X.

smuzaffar committed:
resolve merge conflict for 81x

davidlange6 committed:
Merge pull request #13373 from alexeybaskakov/ComphepSingletopFilterPy8_CMSSW_7_6_X

CompHEP interface to Pythia8

davidlange6 committed:
Merge pull request #13361 from matz-e/fix-calotpg-hcalet

Fix the hcaletValue function.

cmsbuild committed:
Merge pull request #13683 from bsunanda/Run2-hcx74

bsunanda:Run2-hcx74 Add scenario to help testing 2016 HCAL setup

cmsbuild committed:
Merge pull request #13590 from makortel/enablePhase1DqmModules

Enable all working DQM and validation modules for 2017

Commit:
modifier for high pu L1CSC

cmsbuild commented:
Pull request #13243 was updated. @diguida, @cmsdoxy, @cerminar, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @perrotta, @civanch, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @ianna, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @boudoul, @emeschi, @monttj, @srimanob, @hengne can you please check and sign again.

cmsbuild commented:
Pull request #13243 was updated. @diguida, @cmsdoxy, @cerminar, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @perrotta, @civanch, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @ianna, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @boudoul, @emeschi, @monttj, @srimanob, @hengne can you please check and sign again.

Commit:
add muonTrigger2017 to tracking2017 chains

cmsbuild commented:
Pull request #13243 was updated. @diguida, @cmsdoxy, @cerminar, @fabozzi, @Martin-Grunewald, @mommsen, @vanbesien, @perrotta, @civanch, @franzoni, @cmsbuild, @fwyzard, @davidlange6, @smuzaffar, @Dr15Jones, @cvuosalo, @mdhildreth, @ianna, @deguio, @slava77, @ggovi, @vadler, @mmusich, @mulhearn, @boudoul, @emeschi, @monttj, @srimanob, @hengne can you please check and sign again.

davidlange6 commented:
@dildick -looks like you need to rebase this PR now

dildick commented:
Wow, I didn't realize that the commits go into two PRs at the same time. Ok. Will rebase

dildick commented:
I'll have to close this branch, because the Eras.py file in 80X does not have trackingPhase1 in Run2_2017
